include/MeshAmalgamationBase.h

```diff
@@ -30,7 +32,7 @@ class MeshAmalgamation {
   libMesh::LinearImplicitSystem &_system;
   libMesh::DofMap &_dof_map;
 
-  double getElementDataFromMesh(const libMesh::Elem *elem);
+  double getElementDataFromMesh(const libMesh::Elem *elem,const unsigned int variable_index);
```

Comment:
Might be cleaner to add some inline functions:
```suggestion
  double getElementDataFromMesh(const libMesh::Elem *elem,const unsigned int variable_index);
  double getVariableData(const libMesh::Elem *elem) { return getElementDataFromMesh(elem, _variable_index); };
  double getMetricData(const libMesh::Elem *elem) { return getElementDataFromMesh(elem, _metric_index); };

```